src/librustc_data_structures/obligation_forest/mod.rs

```diff
@@ -878,6 +882,8 @@ impl<O: ForestObligation> ObligationForest<O> {
         let mut error_or_done_nodes = mem::take(self.error_or_done_nodes.get_mut());
```

Comment:
Github won't let me attach this comment at the top of this function, but the name of this function is `compress`, and the comment says:

> Compresses the vector, removing all popped nodes \[...]

but is that really true anymore? It seems like we are no longer re-arranging the ordering, but rather using the "dead node" scheme (combined perhaps with node-numbers?) to re-use storage?

Reply:
Yep, corrected the comment.